prac_05/word_occurences.py

```diff
@@ -8,3 +8,56 @@
 - Align the numbers so they are in one nice column (see prac for notes)
 """
 
+from operator import itemgetter
+
+
+def main():
+    # requests phrase from user
+    phrase = input("enter a phrase:")
+    new_dictionary = convert_to_dict(phrase)
+    longest_word = get_longest_word(new_dictionary)
+    sorting_list = initialize_sorting_list(new_dictionary)
+
+    # Sorting the list by key and then printing
+    sorting_list.sort()
+    print("Below is the dictionary sorted by key:")
+    for i in sorting_list:
+        print("{:{}}: {}".format(i[0], longest_word, i[1]))
+
+    #sorting the list by value and then printing
+    sorting_list.sort(key=itemgetter(1), reverse=True)
+    print("Below is the dictionary sorted by value:")
+    for i in sorting_list:
+        print("{:{}}: {}".format(i[0], longest_word, i[1]))
+
+
+# converts the user's string to a dictionary
+def convert_to_dict(string):
+    # place each word into a dictionary with a count of 1, check if it exists, if so add a count
+    dictionary = {}
+    for each in string.split():
+        try:
+            dictionary[each] += 1
+        except KeyError:
+            dictionary[each] = 1
+    return dictionary
+
+
+# determines the longest word in the dictionary, used for printing the output
+def get_longest_word(dictionary):
+    length = 0
+    for word in dictionary:
+        if len(word) > length:
+            length = int(len(word))
+    return length
+
+
+# creates a list from the new_dictionary to use the sort method
+def initialize_sorting_list(dictionary):
```

Comment:
This is unnecessary. Since you only want to sort by the keys, you can just sort the keys and iterate through that, getting the value for each key.
This method would be useful if you wanted to sort by the value, not the key.

Reply:
i think my problem here is i thought we were sorting by the value, then i wasn't sure so did both. But with that said i did use the sorting list for the key, unnecessarily. So this is good feeback! thanks :)